### PR TITLE
test(core): curate format_advice.feature — wire advice CLI contracts, push coverage to lib units

### PR DESCRIPTION
**Pull request by cmbays**
## Summary

Curated BDD pass, **group 11 of 14** (`format_advice`). The advice format is a genuine consumer contract (the `/cut-the-crap` skill ingests it), so the CLI-process behaviors become the acceptance layer; the diagnostic *content* is already owned by `domain::diagnostic` and drops a level.

- **WIRE 8** `@wired` CLI-process contracts in a new `format_advice_cucumber.rs` (single executable Background + 5-fn fixture):
  - the canonical envelope on stdout (`schema_version` 2, `view.shown[]`, JSON-only — no table borders/prose)
  - diagnostic gating: over-threshold verdicts carry the four-field diagnostic, under-threshold ones omit the key (end-to-end through the real walker + coverage + diagnostic engine)
  - exit-code parity with `--format json`
  - the `--no-fail` gate keystone (reports findings, flips only the exit code)
  - diagnostics survive view truncation under `--top`
  - stream separation (`--format advice` → one `[crap=…]` stderr line per exceeder; `--format json` → no diagnostics, no stderr summary)
  - byte-determinism (timestamp aside)

- **DELETE 12** scenarios owned by lower levels (no coverage loss):
  | Deleted | Owner |
  |---|---|
  | SuggestedAction taxonomy (×4: AddTests / Extract / both / AcceptInherent) | `domain::diagnostic::pick_actions_*` |
  | ProposedSplit five-field shape | `domain::diagnostic::proposed_split_round_trips` + serde |
  | Exactly-one-recommended | `dedup_splits_marks_exactly_one_recommended*` |
  | De-dup priority outline | `dedup_splits_keeps_highest_priority*` |
  | `root_cause` derivation outline | `compute_diagnostic_*` / `pick_actions_*` |
  | no-prose / no-names | `domain::diagnostic` serde shape |
  | `--sort-by` / `--min-coverage` composition | `domain::view` sort/filter (+ cli_ergonomics) |
  | `--explain` is NOT advice | complexity_breakdown harness owns `--explain` (and it's prose-stale: `--explain` exits 2 without `--breakdown`) |

- **ABSORB + DELETE** `format_advice_integration.rs` (17 tests).

## Coverage: corrected #415 discipline (the important part)

A normal `#[test]` integration test that **shells the binary** *does* contribute llvm-cov coverage — nextest runs it, and the instrumented subprocess writes profraw that llvm-cov merges. (Only a `harness = false` cucumber runner, which nextest skips, contributes nothing.) Prior shelling-integration deletions passed the gate only because their lib paths were *redundantly* covered by in-process units.

Here two lib functions had **no** in-process coverage and would have spiked past the strict-8 gate on deletion:
- `core::populate_diagnostics` → CRAP 12.17 (44.4%)
- `cli::validate_view_args` → CRAP 17.80 (20%)

Rather than keep a redundant bin-shelling test, I pushed that coverage **down to where it belongs**:
- in-process unit tests for `validate_view_args` (`crap-core/src/cli/view_args.rs`, 5 cases)
- an in-process `analyze()` pipeline test for `populate_diagnostics` (`crap4rs/tests/analyze_pipeline_tests.rs`, enabled + disabled)

The strict-8 self-gate now passes with these functions properly unit-covered:

```
Summary: 1681 functions | 0 above threshold (8) | PASS
```

This PR therefore touches `crap-core/src` (test modules only — no production logic; `wire_envelope_crap4rs` snapshots crap4rs/src wire output and is unaffected).

## Gates (all green)

`cargo fmt --check` · `cargo +1.96.0 clippy --workspace --all-targets -D warnings` · `cargo nextest run --workspace --all-targets` (1339 passed) · `cargo test --test format_advice_cucumber` (8 scenarios / 33 steps) · `bdd-tracked-lint.py` (242 scenarios) · `mutants-skip-lint.py` · `cargo +1.93 check` (MSRV) · `cargo doc -D warnings` · **strict-8 self-gate (0 above threshold)**.

Net **−333** lines.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive unit tests for validation logic with edge-case coverage
  * Implemented behavior-driven development specifications for format validation
  * Created new integration test framework for enhanced end-to-end testing
  * Updated CI pipeline to execute expanded test suites
  * Reorganized test infrastructure for improved maintainability

### CODE REVIEW

File: crates/crap4rs/tests/format_advice_cucumber.rs [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+    assert_ne!(json_cmd, cmd, "command did not contain `--format advice`");
+    let json_run = run(world.require_dir(), &parse_args(&json_cmd));
+    assert_eq!(
+        world.require_output().status.code(),
+        json_run.status.code(),
+        "advice and json exit codes must match\nadvice: {}\njson exit: {:?}",
+        world.fail_context(),
+        json_run.status.code()
+    );
+}
```
[the remark this answers is not in the document]
Comment (cmbays) [reply crates/crap4rs/tests/format_advice_cucumber.rs:258]: Fixed in 5930682 — the exit-parity assertion now includes the `--format json` run's stdout + stderr, so an unexpected json-run failure is actionable rather than opaque.

File: crates/crap4rs/tests/format_advice_cucumber.rs [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+        json_without_timestamp(&world.stdout()),
+        json_without_timestamp(&again_stdout),
+        "advice stdout must be deterministic (timestamp aside)"
+    );
+    assert_eq!(
+        world.stderr(),
+        String::from_utf8_lossy(&again.stderr),
+        "advice stderr summary must be byte-identical across runs"
+    );
+}
```
[the remark this answers is not in the document]
Comment (cmbays) [reply crates/crap4rs/tests/format_advice_cucumber.rs:377]: Fixed in 5930682 — the determinism step now asserts the second run's exit code matches the first (with stdout+stderr context) before parsing its stdout, so a failed/signaled re-run fails fast instead of panicking in the JSON parse. (The timestamp strip already uses a structured `serde_json::Value` parse + `obj.remove("timestamp")`, not string replacement.)
